splice_machine/src/main/java/com/splicemachine/pipeline/foreignkey/ForeignKeyChildInterceptWriteHandler.java

```diff
@@ -159,24 +159,20 @@ public void close(WriteContext ctx) throws IOException {
 
     private boolean hasData(DataResult result,SimpleTxnFilter filter) throws IOException {
         if(result!=null && !result.isEmpty()) {
-            int cellCount = result.size();
             for (DataCell dc : result) {
                 DataFilter.ReturnCode returnCode = filter.filterCell(dc);
                 switch (returnCode) {
-                    case NEXT_ROW:
-                        return false; //the entire row is filtered
-                    case SKIP:
-                    case NEXT_COL:
-                    case SEEK:
-                        cellCount--; //the cell is filtered
-                        break;
-                    case INCLUDE:
+                    case INCLUDE: // fallthrough
                     case INCLUDE_AND_NEXT_COL: //the cell is included, so we have some data
-                    default: //do nothing
+                        return true;
+                    case SKIP: // fallthrough
+                    case NEXT_COL: // fallthrough
+                    case SEEK: // fallthrough
+                    case NEXT_ROW: // fallthrough
+                    default: //do nothing, keep trying
```

Comment:
does it make sense still to use switch here, as most of the cases empty? Maybe "if" will look simpler

Reply:
If it is OK, I would rather keep it this way mainly to illustrate the exact purpose of the algorithm for each `DataFilter.ReturnCode` return type. But I don't have strong opinion about it and can still change it if you think this doesn't make too much sense.